src/handshake.rs

```diff
@@ -403,11 +403,43 @@ pub struct Response {
     status: u16,
     reason: String,
     headers: Vec<(String, Vec<u8>)>,
+    body: Vec<u8>,
 }
 
 impl Response {
     // TODO: resolve the overlap with Request
 
+    pub fn new<R>(status: u16, reason: R) -> Response
+        where R: Into<String>,
+    {
+        Response {
+            status: status,
+            reason: reason.into(),
+            headers: Vec::new(),
+            body: Vec::new(),
+        }
+    }
+
+    /// Get the response body.
+    #[inline]
+    pub fn body(&self) -> &[u8] {
+        &self.body
+    }
+
+    /// Set the response body.
+    #[inline]
+    pub fn set_body(&mut self, body: &[u8]) {
```

Comment:
What do you think about `T: Into<Vec<u8>>`? That would make the caller aware of the required allocation but also allow some flexibility.